datafusion/expr/src/udaf.rs

```diff
@@ -119,9 +119,12 @@ impl AggregateUDF {
     where
         F: AggregateUDFImpl + 'static,
     {
-        Self {
-            inner: Arc::new(fun),
-        }
+        Self::new_from_shared_impl(Arc::new(fun))
+    }
+
+    /// Create a new `AggregateUDF` from a `[AggregateUDFImpl]` trait object
```

Comment:
BTW it might also be nice to add a `From` impl for this  similar to 

https://github.com/apache/datafusion/blob/4f6062e81b7201b58da2b20c63333a5e3ae3c8df/datafusion/expr/src/udaf.rs#L288-L295

That would allow creation using
```rust
let fun: fun: Arc<dyn AggregateUDFImpl> = ...;
// create using `from`:
let new_func = AggregateUDFImpl::from(fun);
```

Reply:
i am not against `AggregateUDFImpl::from` explicit syntax, but i am not a big fan of `.into()` implicit syntax that this unlocks
Into/From is very important for readability in some cases, but it can also [be abused](https://github.com/apache/datafusion/blob/d5ff3e7c247430865590c592587ad60fdff4bcc0/datafusion/common/src/scalar/mod.rs#L3453). Sometimes just single API method is enough

Reply:
To be clear, I am suggesting it in addition to this explicit function. I don't have a strong opinion

Reply:
> To be clear, I am suggesting it in addition to this explicit function. I don't have a strong opinion

i don't very strong either here, just didn't see need for that
let's revisit if needed.

Reply:
thanks for filing that issue! ❤️